src/CPK.ts

```diff
@@ -219,59 +245,59 @@ class CPK {
   async execTransactions(
     transactions: Transaction[],
     options?: ExecOptions
-  ): Promise<TransactionResult> {
-    if (!this.transactionManager) {
-      throw new Error('CPK transactionManager uninitialized')
-    }
-
+  ): Promise<TransactionResult | void> {
     if (this.isSafeApp() && transactions.length >= 1) {
       const standardizedTxs = transactions.map(standardizeSafeAppsTransaction)
-      return this.transactionManager.execTransactions({
-        appsSdk: this.appsSdk,
-        transactions: standardizedTxs
-      })
+      return this.#safeAppsSdkConnector.sendTransactions(standardizedTxs)
     }
 
     if (!this.address) {
       throw new Error('CPK address uninitialized')
     }
-    if (!this.contract) {
+    if (!this.#contract) {
       throw new Error('CPK contract uninitialized')
     }
-    if (!this.masterCopyAddress) {
+    if (!this.#masterCopyAddress) {
       throw new Error('CPK masterCopyAddress uninitialized')
     }
-    if (!this.fallbackHandlerAddress) {
+    if (!this.#fallbackHandlerAddress) {
       throw new Error('CPK fallbackHandlerAddress uninitialized')
     }
-    if (!this.ethLibAdapter) {
+    if (!this.#ethLibAdapter) {
       throw new Error('CPK ethLibAdapter uninitialized')
     }
+    if (!this.#transactionManager) {
+      throw new Error('CPK transactionManager uninitialized')
+    }
 
     const ownerAccount = await this.getOwnerAccount()
+    if (!ownerAccount) {
+      throw new Error('CPK ownerAccount uninitialized')
+    }
+
     const safeExecTxParams = this.getSafeExecTxParams(transactions)
     const sendOptions = normalizeGasLimit({ ...options, from: ownerAccount })
 
-    const codeAtAddress = await this.ethLibAdapter.getCode(this.address)
+    const codeAtAddress = await this.#ethLibAdapter.getCode(this.address)
     const isDeployed = codeAtAddress !== '0x'
 
-    const txManager = !isDeployed ? new CpkTransactionManager() : this.transactionManager
+    const txManager = !isDeployed ? new CpkTransactionManager() : this.#transactionManager
 
     const cpkContracts: CPKContracts = {
-      safeContract: this.contract,
-      proxyFactory: this.proxyFactory,
-      masterCopyAddress: this.masterCopyAddress,
-      fallbackHandlerAddress: this.fallbackHandlerAddress
+      safeContract: this.#contract,
+      proxyFactory: this.#proxyFactory,
+      masterCopyAddress: this.#masterCopyAddress,
+      fallbackHandlerAddress: this.#fallbackHandlerAddress
     }
 
     return txManager.execTransactions({
       ownerAccount,
       safeExecTxParams,
       transactions,
       contracts: cpkContracts,
-      ethLibAdapter: this.ethLibAdapter,
+      ethLibAdapter: this.#ethLibAdapter,
       isDeployed,
-      isConnectedToSafe: this.isConnectedToSafe,
+      isConnectedToSafe: this.#isConnectedToSafe,
       sendOptions
     })
```

Comment:
I would `await` the result and cast it `as TransactionResult`. Note that the only time it would actually be `void` is when connected as a Safe App and transactions length is 0. Maybe add a check for the latter and return an empty object early. Ultimately, it'd be best not to have to cast it here, but that will require some refactoring.